src/modules/75-cf/pages/onboarding/__test__/CreateAFlagView.test.tsx

```diff
@@ -0,0 +1,147 @@
+
+    userEvent.click(selectInput as TargetElement)
+
+    const dropdownOptions = document.getElementsByTagName('li')
+    expect(dropdownOptions).toHaveLength(CF_DEFAULT_PAGE_SIZE)
+    expect(dropdownOptions[0]).toHaveTextContent('hello world')
+  })
+
+  test('It should filter Select options by typed text', async () => {
+    renderComponent()
+
+    const selectInput = document.querySelector('input[id="selectOrCreateFlag"]') as TargetElement
+    expect(document.querySelector('input[id="selectOrCreateFlag"]')).toBeVisible()
+    expect(fetchFlags).toHaveBeenCalled()
+
+    userEvent.click(selectInput)
+
+    const dropdownOptions = document.getElementsByTagName('li')
+    expect(dropdownOptions).toHaveLength(CF_DEFAULT_PAGE_SIZE)
+    expect(dropdownOptions[0]).toHaveTextContent('hello world')
+    expect(dropdownOptions[1]).toHaveTextContent('XYZ Flag')
+    expect(dropdownOptions[2]).toHaveTextContent('ABC Flag')
+
+    userEvent.type(selectInput, 'hello')
+    expect(fetchFlags).toBeCalled()
+    expect(dropdownOptions).toHaveLength(1)
+    expect(dropdownOptions[0]).toHaveTextContent('hello world')
+  })
+
+  test('It should set the selected dropdown option as the selected flag for Onboarding', () => {
+    renderComponent()
+
+    const selectInput = document.querySelector('input[id="selectOrCreateFlag"]')
+    expect(selectInput).toBeVisible()
+    expect(screen.queryByTestId('ffOnboardingSelectedFlag')).not.toBeInTheDocument()
+    expect(fetchFlags).toHaveBeenCalled()
+
+    userEvent.click(selectInput as TargetElement)
+
+    const dropdownOptions = document.getElementsByTagName('li')
+    expect(dropdownOptions).toHaveLength(CF_DEFAULT_PAGE_SIZE)
+    expect(dropdownOptions[0]).toHaveTextContent('hello world')
+
+    userEvent.click(dropdownOptions[0])
+
+    expect(selectInput).toHaveValue('hello world')
+    expect(setSelectedFlag).toBeCalledWith(mockFeatureFlags.features[0])
+  })
+
+  test('It should display a label for the selected flag name and id', async () => {
+    renderComponent({ selectedFlag: mockFeatureFlags.features[0] as any })
+
+    const selectInput = document.querySelector('input[id="selectOrCreateFlag"]') as TargetElement
+    expect(document.querySelector('input[id="selectOrCreateFlag"]')).toBeVisible()
+    expect(fetchFlags).toBeCalled()
+    expect(selectInput).toHaveValue('hello world')
+
+    expect(screen.getByTestId('ffOnboardingSelectedFlag')).toBeVisible()
+    expect(screen.getByTestId('ffOnboardingSelectedFlag').textContent).toMatch(
+      /cf.onboarding.youreUsinghello world.+hello_world.+/g
+    )
+  })
+
+  test('It should create a new flag when search text matches no existing flags', async () => {
+    const createNewFlag = jest.spyOn(ffServices, 'useCreateFeatureFlag')
+    renderComponent()
+
+    const selectInput = document.querySelector('input[id="selectOrCreateFlag"]') as TargetElement
+    userEvent.click(selectInput)
+    userEvent.type(selectInput, 'Onboarding Flag 1')
```

Comment:
`type` inputs each character one-by-one with a slight delay, so the longer a sentence, the longer the test will take to run. Add `allAtOnce: true` to act like it pasted instead.
```suggestion
    userEvent.type(selectInput, 'Onboarding Flag 1', { allAtOnce: true })
```

This seems to be a big cause of test flakiness with timeouts.